pages/rfc/registry-updates.rst

```diff
@@ -27,14 +27,91 @@ DRAFT
 Terminology
 ***********
 
+TODO
+
 - Event: an event in the registry event log, e.g. the RegisterVendor event
 - Event payload: the actual contents of the event, describing an object.
 - Object: the thing an event describes; either a vendor, organization or endpoint.
 - Event header: top-level properties of an event describing its payload.
 
-Challenges
+Operations
 **********
 
+The mutate operations that can be performed (by human operators or systems) on the registry are as follows:
+
+**Vendors**
+
+* Register vendor (currently supported)
+* Update vendor name
+* Reissue CA certificate
+* Rotate keys (and reissue associated certificates)
+* Deactivate/reactivate vendor
+
+**Organizations**
+
+* Register (claim) organization as vendor (currently supported)
+* Update organization name
+* Reissue CA certificate
+* Rotate keys (and reissue associated certificates)
+* Migrate organization to another vendor
+
+**Endpoints**
+
+* Register endpoint for organization (currently supported)
+* Update endpoint (URL, type, properties)
+* Deactivate/reactivate endpoint
+
+.. note::
+
+    There are other operations that are somewhat related to the registry like issuing a TLS certificate to another vendor
+    or revoking a certificate, but these don't mutate the registry and thus aren't relevant for this RFC.
+
+Eventing style
+**************
+
+From an operator's perspective (be it a human or system) they will be executing one of the operations described above.
+However, this isn't necessarily how it's they're mapped to events. These 2 styles have been taken into consideration,
+*domain driven* and *state driven*:
+
+**Domain Driven**
+
+Events are modeled in such a way that they resemble the domain (the operations described earlier) as close as possible
+e.g.: ``RegisterVendor``, ``UpdateVendorName``, ``IssueVendorCertificate``, ``UpdateOrganizationName``,
+``DeactivateVendor``, ``ReactivateVendor``, ``MigrateOrganization``, ``UpdateEndpoint``, etc.
+
+Events itself are very expressive and it is immediately clear what the event's intention. It makes it easy for Nuts nodes
+and auxiliary systems to rationalize or report about the data.
+
+Determining the current state of an object is done by *replaying* all events, from the very first until the very last.
+When there are many events this can be an computationally expensive task. There's also the challenge of legacy: since
+every event is processed again and again long after it was published, every event needs to be supported indefinitely.
+
+.. note::
+
+    Calculating objects' state by replaying cumulative events is known as *event sourcing*.
+
+**State Driven**
```

Comment:
This would allow for garbage collection and only storing hashes but it would require all nodes to agree to a certain hash of cumulative events!